### ISSUE TITLE
Functions ignore the variables before the spread syntax

### ISSUE THREAD

**Issue by gokomer** [vlang/v#8891]
**V version:** 
V 0.2.2 f18adf7

**OS:** 
OS: linux, "Fedora release 33 (Thirty Three)"
Processor: 12 cpus, 64bit, little endian, AMD Ryzen 5 PRO 4650G with Radeon Graphics
CC version: cc (GCC) 10.2.1 20201125 (Red Hat 10.2.1-9)

**What did you do?**

```v

fn sum(a ...int) int {
	mut total := 0
	for x in a {
		total += x
	}
	return total
}

b := [5, 6, 7]
println(sum(3, 5, ...b)) 
```

**What did you expect to see?**
26
 
**What did you see instead?**
18

**Comment by changrui** [comment]
So your `sum func` must defined as `fn sum(a int, b int, ...int)`.

**Comment by dumblob** [comment]
>So your `sum` func must defined as `fn sum(a int, b int, c ...int)`.

This would extremely (and unexpectedly and unnecessarily) limit the use of the spread operator.

The example above is actually two totally separate things:

1. Use of `...` in the declaration: `fn sum( a ...int )` accepting **all** of `sum( 1, 2, 3, 4 )` `sum( 1, 2, ...[3, 4] )` `sum( 1, ...[2, 3], 4 )` `sum( ...[1, 2], 3, 4 )` `sum( ...[1, 2], ...[3, 4] )` `sum( ...[1, 2, 3, 4] )`.
     In other words accept any number of integers **disregarding** how you got them.
2. Use of `...` at the call site:
    2.1. Either with compile-time known values (as shown in (1) above or by having the array `const ( myarray = [1, 2, 3, 4] )`).
    2.2. Or even with runtime spreading (e.g. both `myarray := [1, 2, 3, 4]` or `mut myarray := [1, 2, 3, 4]`).
    In other words spread the given compound structure by one level and first then do any type check.

I'm certain that (1) and (2.1.) is the expected behavior in *V*. But if [any](https://github.com/vlang/v/issues/6853 ) gets implemented, then implementing (2.2.) would also make sense.

**Comment by yuyi98** [comment]
If you have these usages, I think it is best to form an array before calling, I wish Vlang was simple like Golang.
